validation/messages.go

```diff
@@ -0,0 +1,50 @@
+	MsgExpectedRFC3339Date                = "expected string to be RFC 3339 date"
+	MsgExpectedRFC3339Time                = "expected string to be RFC 3339 time"
+	MsgExpectedRFC5322Email               = "expected string to be RFC 5322 email: %v"
+	MsgExpectedRFC5890Hostname            = "expected string to be RFC 5890 hostname"
+	MsgExpectedRFC2673IPv4                = "expected string to be RFC 2673 ipv4"
+	MsgExpectedRFC2373IPv6                = "expected string to be RFC 2373 ipv6"
+	MsgExpectedRFC3986URI                 = "expected string to be RFC 3986 uri: %v"
+	MsgExpectedRFC4122UUID                = "expected string to be RFC 4122 uuid: %v"
+	MsgExpectedRFC6570URITemplate         = "expected string to be RFC 6570 uri-template"
+	MsgExpectedRFC6901JSONPointer         = "expected string to be RFC 6901 json-pointer"
+	MsgExpectedRFC6901RelativeJSONPointer = "expected string to be RFC 6901 relative-json-pointer"
+	MsgExpectedRegexp                     = "expected string to be regex: %v"
+	MsgExpectedMatchSchema                = "expected value to match at least one schema but matched none"
+	MsgExpectedNotMatchSchema             = "expected value to not match schema"
+	MsgExpectedBoolean                    = "expected boolean"
+	MsgExpectedNumber                     = "expected number"
+	MsgExpectedString                     = "expected string"
+	MsgExpectedBase64String               = "expected string to be base64 encoded"
+	MsgExpectedArray                      = "expected array"
+	MsgExpectedObject                     = "expected object"
+	MsgExpectedArrayItemsUnique           = "expected array items to be unique"
+	MsgExpectedOneOf                      = "expected value to be one of \"%s\""
+	MsgExpectedMinimumNumber              = "expected number >= %v"
+	MsgExpectedExclusiveMinimumNumber     = "expected number > %v"
+	MsgExpectedMaximumNumber              = "expected number <= %v"
+	MsgExpectedExclusiveMaximumNumber     = "expected number < %v"
+	MsgExpectedNumberBeMultipleOf         = "expected number to be a multiple of %v"
+	MsgExpectedMinLength                  = "expected length >= %d"
+	MsgExpectedMaxLength                  = "expected length <= %d"
+	MsgExpectedBePattern                  = "expected string to be %s"
+	MsgExpectedMatchPattern               = "expected string to match pattern %s"
+	MsgExpectedMinItems                   = "expected array length >= %d"
+	MsgExpectedMaxItems                   = "expected array length <= %d"
+	MsgExpectedMinProperties              = "expected object with at least %d properties"
+	MsgExpectedMaxProperties              = "expected object with at most %d properties"
+	MsgExpectedRequiredProperty           = "expected required property %s to be present"
+	MsgExpectedDependentRequiredProperty  = "expected property %s to be present when %s is present"
+)
+
+// ErrorFormatter is a function that formats an error message
+var ErrorFormatter func(format string, a ...any) string = fmt.Sprintf
```

Comment:
Random thought: what about making this aware of the request context to enable i18n/l10n scenarios? For example, you could then use a library to return a localized error message based on the request's `Accept-Language` header. As is with your PR you could provide a single langauge, what I'm suggesting here is providing many languages and choosing one based on the request's preferences.

Reply:
Actually it is the next change I want to submit :) But decided to split the work into smaller pieces.

I noticed you try to avoid strings allocation during request validation. Using ctx would break it because we won't be able to do pre-computation.

The current approach I'm trying is to define 2 interfaces:
```go
// ErrorFormatter is a function that formats an error message
var ErrorFormatter func(format string, a ...any) string = fmt.Sprintf

// ErrorFormatterContext is a function that will be used instead of ErrorFormatter
// to format an error message when defined
var ErrorFormatterContext func(ctx Context, format string, a ...any) string = nil
```

Usage of `ErrorFormatterContext` is opt-in. I didn't do benchmarking yet though. If you don't mind I'll continue with it in the following weeks.

Reply:
I also moved it to the root of huma. It would make code easier to read as we introduce context.

Reply:
Good point about the pre-allocation. This probably requires a bit more thought, so I'm happy to get this improvement in first!